rust/gitxetcore/src/xetblob/file_operations/cp_operations.rs

```diff
@@ -63,80 +63,102 @@ async fn build_cp_operation_list(
     for src in sources {
         let src_is_directory;
         let src_path: String;
-
-        if src.contains('*') {
-            return Err(GitXetRepoError::InvalidOperation(
-                "Error: Currently, wildcards are not supported.".to_owned(),
-            ));
-        }
+        let dest_dir: String;
+        let dest_path: String;
+        let mut src_size = 0;
 
         // Validate that the source path is specified correctly, and set src_is_directory.
-        let Some(src_info) = src_fs.info(src).await? else {
-            return Err(GitXetRepoError::InvalidOperation(
-                "Error: Source {src_path} does not exist".to_owned(),
-            ));
-        };
-
-        if src.ends_with('/') {
-            src_path = src.strip_suffix('/').unwrap_or(src.as_str()).to_owned();
 
-            if !src_info.is_dir_or_branch() {
+        if src.contains('*') {
+            // Handling wildcard cases
+            // The file card matching works by recasting the operation as a directory -> directory copy,
+            // but with a possible filter pattern.
+
+            // validate
+            // we only accept globs of the for blah/blah/blah/[glob]
+            // i.e. the glob is only in the last component
+            // src_root_dir should be blah/blah/blah here
+            let src_root_dir = src_fs.parent(src.as_str());
+            
+            if let Some(src_root_dir_path) = src_root_dir {
+                if src_root_dir_path.contains('*') {}
+            }
+            
+            if src_root_dir.is_some_and(|d| d.contains('*'))   {
```

Comment:
These tests just look for the existence of `*`; without implementing glob-style filters, we should test that it's exactly `*`.  Currently, it means that `*.jpg` would behave the same as `*`.

We can either implement any glob using a filter (as in pyxet), or just error out if the last component is not exactly `*` However, in that case, the only real use case would be copying the file contents of a directory non-recursively.  Is that enough?